### PR TITLE
UI sound logic

### PR DESCRIPTION
**Pull request by Eryngii01**
Adjusted the ui-sound-logic branch to include the Wwise sound files and integrate the set up UnityEvents with the sound:
- Created an AudioManager class for holding all the Wwise sound events that subscribes to UnityEvents throughout the game
- Added UI Wwise sound events that work with the menu buttons (hover and click), tile movement and trap selection
- Adjusted the music to pause and unpause when the PauseManager toggles a pauseEvent
- Separated the player sound logic from the PlayerController class into a new PlayerSoundsController class that acts as an intermediate between the PlayerController and AudioManager
- Fixed the build sounds for when the player sets up a trap
- Fixed a minor bug for the AcidPitTrap that passes the SurveyDeployment checks when it should fail and just return

### CODE REVIEW

**Review by eon704 — COMMENTED** [review]
Code wise everything looks good!

Also everything works for the Hen sounds. It seems that UI sounds are not set up yet and the new sound manager has missing events. Not sure if this is intended and/or left for later 😅

File: Assets/Prefabs/UI/MainMenuFINAL.prefab
Comment (eon704) [inline Assets/Prefabs/UI/MainMenuFINAL.prefab:1]: Just to make sure it's noted somewhere, we will need to clean that part up at some point if it is being used 😅
No `FINAL` in the name of files 😅

**Review by eon704 — APPROVED** [review]
A little nitpick, but otherwise LGTM!

**Comment by Eryngii01** [comment]
That's my mistake, I wasn't sure what to name it at first...😅 I renamed the other MainMenu prefab to "MainMenuInDialmeScene" and the one we're using in the MainMenu scene to just "MainMenu". Hopefully that fixes things, and thanks for pointing that out!

**Comment by eon704** [comment]
Feel free to merge whenever, PR looking good!

**Pull request merged**